### ISSUE TITLE
Loss function mismatches

### ISSUE THREAD

**Issue by FarInHeight** [ExplainableML/ProbVLM#8]
Hi!
I'm a computer engineering student from PoliTo and I'm using part of your work as basis for my Master's Thesis.

I have just a couple of questions about the loss function:
1) From the code I see you train the `BayesCap_Net` using `TempCombLoss` for Intra-modal Alignment and Cross-modal Alignment. The `TempCombLoss` inside has a L1 component which is not indicated in the original paper.
Should I assume that you forgot to indicate it in your paper or should I delete the L1 component and train just using the `GenGaussLoss`? 
(I also read the BayesCap paper, where you point out that the L1 component is needed and stress its importance to get better results. The question above it's just to be sure if I need to leave the code as it is or not).
2) In the appendix, it is explained that you use the Taylor-series expansion to make the training more stable. However, from the code linked below
https://github.com/ExplainableML/ProbVLM/blob/cb69f28b1ab23142a1c671e004b09b5cb5d8a204/src/losses.py#L27-L29
I see that you didn't write the $1 - \hat{\beta}$ term that is indicated in the paper (please refer to the red box in the image attached below). Could you please clarify for me why you chose this option, especially for the $- \hat{\beta}$ term?
<img src="https://github.com/user-attachments/assets/4c86372b-1675-4ae8-bc4a-1fcfb095c80b" width="50%" alt="attached image">

Thank you in advance for your reply.

**Comment by udion** [comment]
Hey @FarInHeight,

Great observations!! 

My thoughts below:

1) You noticed it right, the origin of L1 loss is from bayescab paper, you could actually remove it from `TempCombLoss` in ProbVLM. During my experiments, I noticed that training was easier (without tedious hyper-parameter search for optimizer) with the L1 component. it may be possible to get good performance without L1 component, but you may have to tweak hyperparams again.

2) Good eye!! I guess I missed while writing the code, thinking that term is constant (which certainly isn't the case, since it is output of the network) -- I think you should try experimenting including -beta in loss, let me know if you see something interesting!!

**Comment by Issac304** [comment]
the pip list is correct？
